diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -9697,6 +9697,8 @@ static int amdgpu_dm_connector_init_helper(struct amdgpu_device *adev,
 		if (!aconnector->mst_root) {
 			drm_connector_attach_vrr_capable_property(&aconnector->base);
 			drm_connector_attach_passive_vrr_capable_property(&aconnector->base);
+			drm_connector_attach_allm_capable_property(&aconnector->base);
+			drm_connector_attach_allm_mode_property(&aconnector->base);
 		}
 
 		if (adev->dm.hdcp_workqueue)
