diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -208,6 +208,7 @@ int amdgpu_emu_mode;
 uint amdgpu_smu_memory_pool_size;
 int amdgpu_smu_pptable_id = -1;
+int amdgpu_allm_mode = 0;
 /*
  * FBC (bit 0) disabled by default
  * MULTI_MON_PP_MCLK_SWITCH (bit 1) enabled by default
   - With this, for multiple monitors in sync(e.g. with the same model),
@@ -591,6 +592,9 @@ MODULE_PARM_DESC(emu_mode, "Emulation mode, (1 = enable, 0 = disable)");
 module_param_named_unsafe(emu_mode, amdgpu_emu_mode, int, 0444);
 
+MODULE_PARM_DESC(allm_mode, "ALLM via DP mode (0 = off, 1 = HDMI only, 2 = force for DP/PCON)");
+module_param_named_unsafe(allm_mode, amdgpu_allm_mode, int, 0444);
+
 /**
  * DOC: ras_enable (int)
  * Enable RAS features on the GPU (0 = disable, 1 = enable, -1 = auto (default))
  */

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -7256,6 +7256,12 @@ static void fill_stream_properties_from_drm_display_mode(
 
 	stream->output_color_space = get_output_color_space(timing_out, connector_state);
 	stream->content_type = get_output_content_type(connector_state);
+
+	if (stream->signal != SIGNAL_TYPE_HDMI_TYPE_A &&
+	    stream->link->local_sink &&
+	    stream->link->local_sink->edid_caps.allm &&
+	    amdgpu_allm_mode == 2)
+		stream->content_type = DISPLAY_CONTENT_TYPE_GAME;
 }
 
 static void fill_audio_info(struct audio_info *audio_info,
